src/main/java/com/aizistral/nochatreports/mixins/client/MixinChatScreen.java

```diff
@@ -86,7 +86,12 @@ private void onInit(CallbackInfo info) {
 
 								if (ServerSafetyState.getCurrent() != ServerSafetyLevel.SECURE) {
 									ServerSafetyState.setAllowChatSigning(true);
-									ServerSafetyState.updateCurrent(ServerSafetyLevel.INSECURE);
+									if(ServerSafetyState.getCurrent() == ServerSafetyLevel.INSECURE){
+										ServerSafetyState.updateCurrent(ServerSafetyLevel.INSECURE);
+									}
```

Comment:
I don't quite follow the logic here, why call `updateCurrent` if state is already set to `INSECURE`?

Reply:
Honestly, I don't know. 
I just looked at how you did it at the moment and thought that maybe I would need to keep the line to make it "re-updated to reflect the state" or something. Can be removed if that's not the case.

Reply:
Yeah that's not needed, just remove it.